applications/HDF5Application/CMakeLists.txt

```diff
@@ -96,7 +94,11 @@ include_directories(${HDF5_INCLUDE_DIRS})
 add_definitions(${HDF5_DEFINITIONS})
 
 add_library(KratosHDF5Core SHARED ${KRATOS_HDF5_APPLICATION_CORE} ${KRATOS_HDF5_APPLICATION_TESTING_SOURCES})
-target_link_libraries(KratosHDF5Core PUBLIC KratosCore PRIVATE ${HDF5_C_LIBRARIES} ${KRATOS_HDF5_APPLICATION_EXTRA_LIBS})
+if(USE_MPI)
+  target_link_libraries(KratosHDF5Core PUBLIC KratosCore KratosMPICore PRIVATE ${HDF5_C_LIBRARIES} ${KRATOS_HDF5_APPLICATION_EXTRA_LIBS})
+else(USE_MPI)
+  target_link_libraries(KratosHDF5Core PUBLIC KratosCore PRIVATE ${HDF5_C_LIBRARIES} ${KRATOS_HDF5_APPLICATION_EXTRA_LIBS})
+endif(USE_MPI)
```

Comment:
I recommend splitting this into a separate MPI-Extension, same as the other apps

Reply:
Would it be ok if I do it in a seperate PR? I think that will be cleaner, because there are some other files which also needs to go into this mpi-extension.

Reply:
ok for me